### DISCUSSION TITLE
Security concern on InboxSDK App Id

### DISCUSSION THREAD

**Discussion by HbZh** [InboxSDK/InboxSDK#928] [General]
**Question(s)**

Hi team, 

What is the purpose to have the App id when I load InboxSDK? What risks am I going to face if I lost my app id and someone get it and implement something with a bad idea. 

Thanks!

**Comment by aleemstreak** [comment]
The purpose of the app ID is for us to be able to tie a particular
extension with its author. Since we have anonymized telemetry coming from
the all extensions (to help detect gmail changes and other issues) we want
to be able to reach out to a developer if we notice they are running into a
particular issue or on an old version of the InboxSDK. To maintain
compatibility between multiple extensions, we need extension developers to
use a relatively recent version of the InboxSDK.

We're open to changing this system but the goal is be able to reach
extension developers.
ᐧ

**Comment by loquatv2** [comment]
> Since we have anonymized telemetry coming from the all extensions

@aleemstreak Could you please be more specific about how this is done? Do I need to include Streak in the host_permissions? I don't think that I've ever seen any network traffic including my AppId. Or does it only happen when something goes wrong? Like @HbZh I do find it confusing because the only mentioning of AppId is in the [this](https://github.com/InboxSDK/InboxSDK/blob/main/README.md#fixing-unreproducible-bugs) section of the ReadMe. It'd be very useful if this repo or the website for registering AppId could include some explanations. Thanks in advance! And thanks for making InboxSDK open source.

**Comment by aleemstreak** [comment]
Good point, we should document more why we need it. 

You don't need to add anything to the host permissions, it logs directly to inboxsdk.com with simple get requests...

**Comment by starostin** [comment]
As source code is open source now, developers can remove part with AppId i think (it is not very difficult to find it). Will it break something? Or there any other restrictions why devs can't do that? 
And other question, is it possible that one day library maintainers will do something with AppId which apps are using, and it will impact extension behavior somehow? (Sorry, but have to ask this, as it could be a huge potential risk)

**Reply by aleemstreak** [reply]
As Fred mentioned the app id isn't really a restriction. You have two options:
1. use an opaque id
2. submit a PR that adds an option to opt out of providing an app id (knowing that enabling this will opt you out of notifications of incompatibilities with Gmail or other extensions)
